x-pack/plugins/security_solution/common/endpoint/generate_data.ts

```diff
@@ -466,11 +493,28 @@ export class EndpointDocGenerator {
       },
       host: this.commonInfo.host,
       process: {
+        pid: options.pid ? options.pid : this.randomN(5000),
+        executable: `C:\\${processName}`,
+        args: `"C:\\${processName}" \\${this.randomString(3)}`,
+        code_signature: {
+          status: 'trusted',
+          subject_name: 'Microsoft',
+        },
+        hash: { md5: this.seededUUIDv4() },
         entity_id: options.entityID ? options.entityID : this.randomString(10),
-        parent: options.parentEntityID ? { entity_id: options.parentEntityID } : undefined,
-        name: options.processName ? options.processName : randomProcessName(),
+        parent: options.parentEntityID
+          ? {
+              entity_id: options.parentEntityID,
+              pid: options.parentPid ? options.parentPid : this.randomN(5000),
+            }
+          : undefined,
+        name: processName,
       },
-    };
+      user: {
+        domain: this.randomString(10),
+        name: this.randomString(10),
+      },
+    } as EndpointEvent;
```

Comment:
why is this cast required? Let's see if we can get rid of it, and if not, please add a comment explaining why it is needed.

Reply:
https://github.com/elastic/kibana/pull/69098/commits/ef3a5394196bce4de4e6b1f7902e44649a4cb5e5